managedtenants/core/addons_loader/addon.py

```diff
@@ -93,12 +97,26 @@ def load_metadata(self, environment):
 
         self._validate_schema(metadata)
         self._validate_extra_resources(environment, metadata)
+        self._validate_subscription_config(metadata)
 
         if "extraResources" in metadata:
             self.extra_resources_loader = FileSystemLoader(str(metadata_dir))
 
         return metadata
 
+    def _validate_subscription_config(self, metadata):
+        if not metadata.get("config"):
+            return
+        configs_present = set(metadata["config"].keys())
```

Comment:
`metadata["config"] `is a dict, no need to call `keys()`, the dict iterator returns the keys anyway.

Reply:
why do you need to deduplicate it? the schema wont allow the same key twice.

Reply:
@apahim I just wanted to create a set out of the keys, so that i could check if its a subset of the allowed keys.